### ISSUE TITLE
eager mode tests started to fail from build 410

### ISSUE THREAD

**Issue by powderluv** [llvm/torch-mlir#789]
https://github.com/llvm/torch-mlir/runs/6149438281?check_suite_focus=true

```
FAIL: TORCH_MLIR_PYTHON :: eager_mode/annotate_args.py (2 of 11)
******************** TEST 'TORCH_MLIR_PYTHON :: eager_mode/annotate_args.py' FAILED ********************
Script:
--
: 'RUN: at line 6';   /opt/hostedtoolcache/Python/3.9.12/x64/bin/python /home/runner/work/torch-mlir/torch-mlir/python/test/eager_mode/annotate_args.py | /home/runner/work/torch-mlir/torch-mlir/build/bin/FileCheck /home/runner/work/torch-mlir/torch-mlir/python/test/eager_mode/annotate_args.py
--
Exit Code: 1

Command Output (stderr):
--
/home/runner/work/torch-mlir/torch-mlir/python/test/eager_mode/annotate_args.py:19:10: error: CHECK: expected string not found in input
# CHECK: Torch Tensor (shape=(1, 3, 32, 32), dtype=torch.float32)
         ^
<stdin>:1:1: note: scanning from here
FAIL - simple
^
<stdin>:2:12: note: possible intended match here
Errors: Couldn't normalize args and kwargs
           ^

Input file: <stdin>
Check file: /home/runner/work/torch-mlir/torch-mlir/python/test/eager_mode/annotate_args.py

-dump-input=help explains the following input dump.

Input was:
<<<<<<
            1: FAIL - simple 
check:19'0     X~~~~~~~~~~~~~ error: no match found
            2: Errors: Couldn't normalize args and kwargs 
check:19'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:19'1                ?                                possible intended match
            3:  
check:19'0     ~
            4: FAIL - handle_zero_dim 
check:19'0     ~~~~~~~~~~~~~~~~~~~~~~~
            5: Errors: Couldn't normalize args and kwargs 
check:19'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            6:  
check:19'0     ~
            7: FAIL - correctly_order_kwargs 
check:19'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            .
            .
            .
>>>>>>

--

********************
Testing:  0.. 10.. 20.
FAIL: TORCH_MLIR_PYTHON :: eager_mode/normalize_args_kwargs.py (4 of 11)
******************** TEST 'TORCH_MLIR_PYTHON :: eager_mode/normalize_args_kwargs.py' FAILED ********************
Script:
--
: 'RUN: at line 6';   /opt/hostedtoolcache/Python/3.9.12/x64/bin/python /home/runner/work/torch-mlir/torch-mlir/python/test/eager_mode/normalize_args_kwargs.py | /home/runner/work/torch-mlir/torch-mlir/build/bin/FileCheck /home/runner/work/torch-mlir/torch-mlir/python/test/eager_mode/normalize_args_kwargs.py
--
Exit Code: 1

Command Output (stderr):
--
/home/runner/work/torch-mlir/torch-mlir/python/test/eager_mode/normalize_args_kwargs.py:15:10: error: CHECK: expected string not found in input
# CHECK: PASS - should_normalize
         ^
<stdin>:1:1: note: scanning from here
FAIL - should_normalize
^

Input file: <stdin>
Check file: /home/runner/work/torch-mlir/torch-mlir/python/test/eager_mode/normalize_args_kwargs.py

-dump-input=help explains the following input dump.

Input was:
<<<<<<
          1: FAIL - should_normalize 
check:15     X~~~~~~~~~~~~~~~~~~~~~~~ error: no match found
          2: Errors: Couldn't normalize args and kwargs 
check:15     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          3:  
check:15     ~
          4: FAIL - shouldnt_normalize1 
check:15     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
          5: Errors: Couldn't normalize args and kwargs 
check:15     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          6:  
check:15     ~
          .
          .
          .
>>>>>>

--

********************
Testing:  0.. 10.. 20.. 30.. 40.. 50.. 60.. 70.. 80.. 90.. 
********************
Failed Tests (2):
  TORCH_MLIR_PYTHON :: eager_mode/annotate_args.py
  TORCH_MLIR_PYTHON :: eager_mode/normalize_args_kwargs.py


Testing Time: 5.31s
  Passed: 9
  Failed: 2
FAILED: tools/torch-mlir/python/test/CMakeFiles/check-torch-mlir-python /home/runner/work/torch-mlir/torch-mlir/build/tools/torch-mlir/python/test/CMakeFiles/check-torch-mlir-python 
cd /home/runner/work/torch-mlir/torch-mlir/build/tools/torch-mlir/python/test && /opt/hostedtoolcache/Python/3.9.12/x64/bin/python /home/runner/work/torch-mlir/torch-mlir/build/./bin/llvm-lit -sv /home/runner/work/torch-mlir/torch-mlir/build/tools/torch-mlir/python/test
ninja: build stopped: subcommand failed.
```

**Comment by silvasean** [comment]
Hey Maks, after you debug this, can you explain why this broke? I'm curious if it is upstream version drift/etc. and how we can prevent it in the future.

**Comment by makslevental** [comment]
@silvasean 

This is drift because I couldn't repro until I updated PyTorch from `1.12.0.dev20220325` to `1.12.0.dev20220425`. Still investigating though.

**Comment by makslevental** [comment]
Fixed in https://github.com/llvm/torch-mlir/pull/793.

@silvasean @powderluv 
The reason for the fail was an upstream change to how [operator_schemas.normalize_function works](https://github.com/pytorch/pytorch/commit/7a80fc2ce7568212b4620ef014f9a70f2efc0772#diff-e7602003c853555c44d805885670cf3500aefe9b8cfec7f8571e2d11f57fd56dR260) (i.e., before this change we didn't that `if` and after that change we do take that `if`), so the fix was simple enough to implement but "weedy" to explain:

`__torch_dispatch__` receives `func: OpOverload` (defined [here](https://github.com/pytorch/pytorch/blob/f4200600e44d8696134b14aac142cfc20b276e36/torch/_ops.py#L30)), which has a pointer to [OpOverloadPacket](https://github.com/pytorch/pytorch/blob/f4200600e44d8696134b14aac142cfc20b276e36/torch/_ops.py#L72). I don't completely understand the reasoning behind the factoring but it's the `OpOverloadPacket` that has the op schema and a pointer to `PyCapsule` object (under the `.op` property). This capsule object is the callable that passes `isinstance(target, types.BuiltinFunctionType)`. This code worked before because they were checking for that invariant (that the object passed to `normalize_function` was a "builtin") in a different way (`target not in boolean_dispatched or target.__module__ not in ['torch.nn.functional', 'torch.functional']`).

The answer to the implicit question of how to avoid this in the future is that there's no good answer - the APIs are explicitly marked as [is_backward_compatible=False](https://github.com/pytorch/pytorch/blob/7a80fc2ce7568212b4620ef014f9a70f2efc0772/torch/fx/operator_schemas.py#L230). We could duplicate their code but that's about it.
